app/hooks/use-countdown/use-countdown.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+import timeDiff from '~/utils/time-diff';
+import now from '~/utils/now';
+import addSeconds from '~/utils/add-seconds';
+
+import type { UseCountdownArgs } from './types'
+
+const useCountdown = ({ target, diffSeconds = 0 }: UseCountdownArgs) => {
```

Comment:
new hook to configure countdown timers, I'm using it on the verification code screen